Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs

```diff
@@ -45,566 +49,256 @@ public void TestInitialize()
-            var expectedResourceName = GetResourceNameForStoredProcedure(command);
-            Assert.IsNull(operationReturned, "Operation returned should be null as all context is maintained internally for Async calls");
-        }
 
-        /// <summary>
-        /// Validates SQLProcessingProfiler sends correct telemetry on calling OnBeginForBeginExecuteReaderInternal.
-        /// </summary>
-        [TestMethod]
-        [Description("Validates sqlProcessingProfiler sends correct telemetry on calling OnBeginForBeginExecuteReaderInternal.")]
-        [Owner("cithomas")]
-        [TestCategory("CVT")]
-        public void RddTestSqlProcessingProfilerBeginAndForBeginExecuteReaderInternal()
-        {
-            var command = GetSqlCommandTestForStoredProc();
-            var returnObjectPassed = new object();
-            this.sqlProcessingProfiler.OnBeginForBeginExecuteReaderInternal(command, null, null, null, null, null);
-            Thread.Sleep(SleepTimeMsecBetweenBeginAndEnd);
-            var objectReturned = this.sqlProcessingProfiler.OnEndForSqlAsync(null, returnObjectPassed, command, null);
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+            var context = this.sqlProcessingProfiler.OnBeginForTwoParameters(command, null);
+            this.sqlProcessingProfiler.OnExceptionForTwoParameters(
+                context,
+                TestUtils.GenerateSqlException(10),
+                command,
+                null);
+            stopwatch.Stop();
 
-            Assert.AreSame(returnObjectPassed, objectReturned, "Object returned from OnEndForSqlAsync processor is not the same as expected return object");
             Assert.AreEqual(1, this.sendItems.Count, "Only one telemetry item should be sent");
             ValidateTelemetryPacket(
                 this.sendItems[0] as DependencyTelemetry,
-                GetResourceNameForStoredProcedure(command),
-                RemoteDependencyConstants.SQL,
-                true,
-                SleepTimeMsecBetweenBeginAndEnd, 
-                "0");
+                expectedName: GetResourceNameForStoredProcedure(command),
+                expectedSuccess: false,
+                expectedResultCode: "10",
+                timeBetweenBeginEndInMs: stopwatch.ElapsedMilliseconds);
         }
 
-        /// <summary>
-        /// Validates SQLProcessingProfiler sends correct telemetry on calling OnExceptionForExecuteReader.
-        /// </summary>
         [TestMethod]
-        [Description("Validates SQLProcessingProfiler sends correct telemetry on calling OnExceptionForExecuteReader.")]
-        [Owner("cithomas")]
-        [TestCategory("CVT")]
-        public void RddTestSqlProcessingProfilerBeginAndForBeginExecuteReaderInternalException()
+        public void OnExceptionSendsCorrectTelemetry3()
```

Comment:
Is there anything more we can add to distinguish these test names with instead of 1, 2, 3?